### PR TITLE
Group moderators edit category description

### PR DESCRIPTION
**Pull request by jbrw**

### CODE REVIEW

File: test/javascripts/acceptance/post-admin-menu-test.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ QUnit.test("Enter as a anon user", async assert => {
   await click(".show-more-actions");
 
   assert.ok(exists("#topic"), "The topic was rendered");
+  assert.ok(exists("#post_1 .post-controls .edit"), "The edit button was not rendered");
```
Comment (eviltrout) [inline test/javascripts/acceptance/post-admin-menu-test.js:10]: I didn't see any JS changed but the tests were changed?

Comment (jbrw) [reply test/javascripts/acceptance/post-admin-menu-test.js:10]: https://github.com/discourse/discourse/blob/032703e6e21df3754a429f839e364a4396d8371b/lib/guardian/post_guardian.rb#L164-L166

An extra condition was added to the can_edit_post? method, which ends up being used by the PostSerializer. So I added an extra assertion here to ensure the button was being rendered as expected, based on the value from the serializer.

Comment (eviltrout) [reply test/javascripts/acceptance/post-admin-menu-test.js:10]: That is cool, just noting that it worked before this commit.

Comment (jbrw) [reply test/javascripts/acceptance/post-admin-menu-test.js:10]: Absolutely - just making sure I didn't introduce any regressions.

File: spec/components/post_revisor_spec.rb [hunk 2]
```diff
@@ -598,6 +598,29 @@
       end
     end
 
+    context "logging group moderator edits" do
+      fab!(:group_user) { Fabricate(:group_user) }
+      fab!(:category) { Fabricate(:category, reviewable_by_group_id: group_user.group.id) }
```
Comment (tgxworld) [inline spec/components/post_revisor_spec.rb:603]:
```suggestion
      fab!(:category) { Fabricate(:category, reviewable_by_group_id: group_user.group.id, topic: topic) }
```

File: spec/components/post_revisor_spec.rb [hunk 3]
```diff
@@ -598,6 +598,29 @@
...
+
+      before do
+        SiteSetting.enable_category_group_moderation = true
+        category.update(topic: topic)
```
Comment (tgxworld) [inline spec/components/post_revisor_spec.rb:607]:
```suggestion
```

File: spec/components/post_revisor_spec.rb [hunk 4]
```diff
@@ -598,6 +598,29 @@
...
+        topic.update(category: category)
+        post.update(topic: topic)
```
Comment (tgxworld) [inline spec/components/post_revisor_spec.rb:609]:
```suggestion
        topic.update!(category: category)
        post.update!(topic: topic)
```

File: spec/requests/posts_controller_spec.rb [hunk 5]
```diff
@@ -426,6 +426,53 @@
...
+        second_topic_in_category = Fabricate(:topic, category: category)
+        post_in_second_topic = Fabricate(:post, user: user, topic: second_topic_in_category)
+
+        put "/posts/#{post_in_second_topic.id}.json", params: update_params
+        expect(response.status).to eq(403)
+      end
+
+      it "can not update category descriptions in other categories" do
+        second_category = Fabricate(:category)
+        topic.update(category: second_category)
```
Comment (tgxworld) [inline spec/requests/posts_controller_spec.rb:468]:
```suggestion
        topic.update!(category: second_category)
```

**Comment by tgxworld** [comment]
@jbrw Some minor suggestions but the PR looks good to me :+1:

**Pull request merged**